src/commands/cmd_explain.c

```diff
@@ -68,8 +68,12 @@ void _MGraph_Explain(void *args) {
 	if(graphname) {
 		gc = GraphContext_Retrieve(ctx, graphname, true);
 		if(!gc) {
-			RedisModule_ReplyWithError(ctx, "key doesn't contains a graph object.");
-			goto cleanup;
+			gc = GraphContext_New(ctx, qctx->graphName, GRAPH_DEFAULT_NODE_CAP, GRAPH_DEFAULT_EDGE_CAP);
+			if(!gc) {
+				CommandCtx_ThreadSafeContextUnlock(qctx);
+				RedisModule_ReplyWithError(ctx, "Graph name already in use as a Redis key.");
+				goto cleanup;
+			}
```

Comment:
@swilly22 Do we want to allow key creation with the EXPLAIN endpoint? This logic is valid, but EXPLAIN can be used without a key to get a default execution plan.

Also, EXPLAIN commands don't get replicated, though ostensibly read-only queries don't either. I think replication doesn't actual matter here, as the first query that actually creates data will cause the graph key to be created in the replicated call.

Reply:
@jeffreylovitz, as you've put it, the first **write** query will be replicated, I don't see an issue with read queries which do not present any nodes/edges to create the graph object on one machine while not being replicated. in case of `graph.explain` I don't have an issue with this command creating a graph object in case a graph key is specified.

Reply:
:+1: